Source/WebCore/platform/graphics/mac/controls/ImageControlsButtonMac.mm

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "ImageControlsButtonMac.h"
+
+#if PLATFORM(MAC) && ENABLE(SERVICE_CONTROLS)
+
+#import "GraphicsContext.h"
+#import "ImageControlsButtonPart.h"
+#import <pal/spi/mac/NSServicesRolloverButtonCellSPI.h>
+
+namespace WebCore {
+
+ImageControlsButtonMac::ImageControlsButtonMac(ImageControlsButtonPart& owningPart, ControlFactoryMac& controlFactory, NSServicesRolloverButtonCell *servicesRolloverButtonCell)
+    : ControlMac(owningPart, controlFactory)
+    , m_servicesRolloverButtonCell(servicesRolloverButtonCell)
+{
+}
+
+IntSize ImageControlsButtonMac::servicesRolloverButtonCellSize()
+{
+    auto& controlFactory = static_cast<ControlFactoryMac&>(ControlFactory::sharedControlFactory());
```

Comment:
Consider adding a `sharedControlFactory` static method on `ControlFactoryMac`, rather than performing the `static_cast` here, similar to `RenderThemeCocoa`.

Reply:
Fixed.